### DISCUSSION TITLE
Making changes to QML files without rebuilding the entire project?

### DISCUSSION THREAD

**Discussion by azzamsa** [KDAB/cxx-qt#477] [Q&A]
Hi.

Every time I made changes to the `*.qml` files, I need to run `cargo clean && cargo run`. Otherwise, the changes don't appear. The build itself takes so long thus it slows my iterations. 

Is there any way to make changes to `*.qml` without rebuilding the entire project?

**Comment by ahayzen-kdab** [comment]
This is a known issue with the cargo build route, see https://github.com/KDAB/cxx-qt/issues/469 . For now just `touch /path/to/qml.qrc` before you build and it'll work.

**Reply by ahayzen-kdab** [reply]
BTW, this should now be in 0.5.1, let us know if there are still issues!

**Comment by alex13sh** [comment]
You can download a qml file from the file system, not from qrc. Thus, changing the qml file will not cause the project to be rebuilt. This is convenient for prototyping.
To do this, you need to remove one line from build.rs
`//         .qrc("qml/qml.qrc")`

And in main.rs specify the path to the qml file relative to the project directory
```
let root = env!("CARGO_MANIFEST_DIR");
engine.load(&QUrl::from(&format!("{root}/qml/main.qml")))
```
